pkg/volume/util/operationexecutor/fakegenerator.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operationexecutor
+
+import (
+	"time"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/pkg/util/mount"
+	"k8s.io/kubernetes/pkg/volume"
+	volumetypes "k8s.io/kubernetes/pkg/volume/util/types"
+)
+
+// fakeOGCounter is a simple OperationGenerator which counts number of times a function
+// has been caled
+type fakeOGCounter struct {
```

Comment:
The existing operationGenerator fake is for internal testing because it allows us to pause and unpause operations etc. But the interface is not easy to use for external testing (IMO). I thought about extending existing functions but it seems ugly and I think we are better off having a fake operation generator that can be used by external controller tests more easily..